pykokkos/interface/views.py

```diff
@@ -704,6 +706,53 @@ def from_cupy(array) -> ViewType:
 
     return from_numpy(np_array, memory_space, layout)
 
+def is_array(array) -> bool:
+    """
+    Check if an object conforms to enough numpy array standards to be treated as an array
+    Using the function from_cupy() as a standard for bare minimum needed to
+    to convert any array into a numpy array
+
+    :param array: the array of unknown type
+    :returns: a true/false if object is an array
+    """
+    
+    required_attr = ["dtype", "data", "shape", "flags"]
+    test_attr = dir(array)
+
+    if(not set(required_attr).issubset(set(test_attr))):
+      return False
+    
+    for d in required_attr:
+      if callable(getattr(array, d, None)):
+        return False
+
+      return True
+
+def from_array(array, space: Optional[MemorySpace] = None, layout: Optional[Layout] = None) -> ViewType:
+    """
+    Create a PyKokkos View from a generic array
+
+    :param array: the array of unknown type
+    :param space: an optional argument for memory space (used by from_cupy)
+    :param layout: an optional argument for layout (used by from_cupy)
+    :returns: a PyKokkos View wrapping the array
+    """
+
+    if isinstance(array, np.ndarray):
+      return from_numpy(array, space, layout)
+    # test if the input array can duck-type to a
+    # numpy array sufficient to our purpose
+    if is_array(array):
+      return from_cupy(array)
```

Comment:
This logic feels a little weird to me. We're saying if it's a numpy array, call from numpy. But if it's an array, we call from_cupy? Why?

Reply:
the from_cupy function is actually a general function, not actually cupy specific. If the array is sufficiently numpy-like, without being numpy, the from_cupy function in theory can handle it.

Reply:
Yeah... the naming initial confused me too, had to go through the function comments to actually see what was going on. I think its ever actually used for cupy so in practice its fine but theoretically is can be used for other array types.

Reply:
Please include necessary comments.

Reply:
I can rename from_cupy to from_array as that s more accurate?